test/e2e-tests/tests/pipelines_test.go

```diff
@@ -0,0 +1,156 @@
+			}
+
+			for _, condition := range pipelineRun.Status.Conditions {
+				if condition.Reason == "Failed" {
+					return fmt.Errorf("pipelineRun \"%v\" failed with message \"%v\"", pipelineRun.Name, condition.Message)
+
+				} else if condition.Reason == "Running" {
+					time.Sleep(time.Second * 5)
+					continue CheckRuns // it is still running so restart the loop and wait 5 seconds
+				}
+			}
+		}
+
+		break // none failed and none were still running so all pipeline runs have complete
+	}
+
+	return nil
+}
+
+// init is called before any test. This is used to build
+// then apply the kustomize config for the main pipeline
+func init() {
+
+	resourceMap, err := support.ApplyPipelineKustomizeFolder(PipelineRelativePath)
+	if err != nil {
+		panic(fmt.Sprintf("error while building kustomize : %v", err.Error()))
+	}
+
+	ctx := CreateContext()
+
+	clients, err := support.CreateClients("test-namespace")
+	if err != nil {
+		panic(fmt.Sprintf("error while creating client : %v", err.Error()))
+	}
+
+	for _, resource := range resourceMap.Resources() {
+		err = support.CreateObjectFromResource(ctx, &clients, resource)
+		if err != nil {
+			panic(fmt.Errorf("error while creating objects from kustomize resources : %v", err.Error()))
+		}
+	}
+}
+
+func Test_PipelineExists(t *testing.T) {
+	t.Skip()
+	ctx := CreateContext()
+
+	clients, err := support.CreateClients("test-namespace")
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	list, err := clients.Pipeline.List(ctx, k8sApi.ListOptions{})
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	expectedPipelineNames := []string{"aiedge-e2e"}
+	if len(list.Items) != len(expectedPipelineNames) {
+		t.Fatalf("expected %v pipelines but %v existed on the cluster", len(expectedPipelineNames), len(list.Items))
+	}
+
+OUTER:
+	for _, name := range expectedPipelineNames {
+		for _, pipeline := range list.Items {
+			if pipeline.Name == name {
+				continue OUTER
+			}
+		}
+
+		t.Fatalf("no pipeline had the name %v", name)
+	}
+}
+
+func Test_DefaultPipelineRun(t *testing.T) {
+	ctx := CreateContext()
+
+	clients, err := support.CreateClients("test-namespace")
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	pipelineRun, err := support.ReadFileAsPipelineRun(PipelineRunRelativePath)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	options, err := support.GetOptions()
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	// setting these values to the options passed in as env vars
+	support.SetPipelineRunParam("s3-bucket-name", support.NewStringParamValue(options.S3BucketName), &pipelineRun)
+	support.SetPipelineRunParam("target-imagerepo", support.NewStringParamValue(options.RegistryRepoName), &pipelineRun)
+
+	_, err = clients.PipelineRun.Create(ctx, &pipelineRun, k8sApi.CreateOptions{})
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	err = WaitForAllPipelineRunsToComplete(ctx, &clients)
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+}
+
+func Test_PipelineRunsCompleted(t *testing.T) {
+	t.Skip()
```

Comment:
I guess this is because of work in progress?

Reply:
I actually just removed the extra tests, nether were needed any more and or did the same as the current test